crates/libm-test/src/mpfloat.rs

```diff
@@ -0,0 +1,320 @@
+//! Interfaces needed to support testing with multi-precision floating point numbers.
+//!
+//! Within this module, the macros create a submodule for each `libm` function. These contain
+//! a struct named `Operation` that implements [`MpOp`].
+
+use std::ops::RemAssign;
+
+use az::Az;
+use rug::ops::PowAssign;
+use rug::Assign;
+pub use rug::Float as MpFloat;
+
+use crate::Float;
+
+/// Create a multiple-precision float with the correct number of bits for a concrete float type.
+fn new_mpfloat<F: Float>() -> MpFloat {
+    MpFloat::new(F::SIGNIFICAND_BITS + 1)
+}
+
+/// Set subnormal emulation and convert to a concrete float type.
+fn prep_retval<F: Float>(mp: &mut MpFloat) -> F
+where
+    for<'a> &'a MpFloat: az::Cast<F>,
+{
+    mp.subnormalize_ieee();
```

Comment:
Oh good catch. Is there a point in asserting that the rounding returned by `subnormalize_ieee_round` is `Equal`?

Reply:
For my clarity, is it correct the result will be correctly rounded to 1/2 ULP regardless of what the return value is? Meaning the returned `Ordering` provides information about what was truncated rather than about accuracy.

Reply:
Changed to use `_round` rather than `_mut` and pass the result to `prep_retval`.